api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -362,6 +362,19 @@ const (
 	// a process to check if the different components in the DataPlane are working as expected. Checks:
 	// - Validates the monitoring stack is properly working after restoration, if not HCCO will restart the prometheus-k8s pods.
 	HostedClusterRestoredFromBackupAnnotation = "hypershift.openshift.io/restored-from-backup"
+
+	// HostedClusterSourcedAnnotation is set to true on Secret and ConfigMap resources to designate them as
+	// hosted-cluster-sourced resources. This means that the hosted cluster version of these resources in the source of
+	// truth and the management cluster version will be just empty resources that have this annotation. This is useful
+	// to enable day-two configuration use cases where such resources are expected to be provided by the end-user after
+	// the cluster creation, and , duo to certain restrictions, those resources include sensitive data that can't live
+	// on the control-plane. Setting this annotation will instruct HyperShift to skip creating this resource on the hosted
+	// cluster and to not override any changes done later on the hosted cluster version of this resource.
+	//
+	// This annotation can only be set on empty resources and currently it'll only be honored when set on secrets that
+	// referenced in the HostedCluster `spec.configuration.authentication.oidcProviders[*].oidcClients[*].clientSecret`
+	// and only for the ARO-HCP platform.
```

Comment:
> only for the ARO-HCP platform

I supposed it will be honored on all platforms as I don't see this enforced in the code. This probably just informs the user where this might be used? Maybe I'd leave this out or rephrase it a little bit.

Reply:
nice catch! actually it's intended to be supported only on ARO-HCP and there was a check in the code that I removed during testing. will put back.